### PR TITLE
feat(table): add `hasFastSearch` prop to toggle onChange/onBlur

### LINKED ISSUE #2871: [StatefulTable] prop to trigger search on 'Enter' instead of as-typing
### What package is this for?

- [X] React
- [ ] Angular

### Summary

Add new prop to only trigger searching on 'Enter' key instead of as-typing for performance sensitive environments

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #2871 

**Summary**

- add `hasFastSearch` prop to toggle search happening onChange or on `Enter` or onBlur.

**Change List (commits, features, bugs, etc)**

- add `hasFastSearch` prop to table
- set `hasFastSearch` to default to `true` to keep current behavior
- pass to TableToolbar
- add onKeyDown/onBlur handlers when `hasFastSearch:false`
- add cypress tests to cover

**Acceptance Test (how to verify the PR)**

- in the [simple stateful table story](https://deploy-preview-2874--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table-statefultable--simple-stateful-example), if you uncheck `hasFastSearch` the table should only update it's state when you press `Enter` or change focus away from the search box

**Regression Test (how to make sure this PR doesn't break old functionality)**

- existing search functionality should be unchanged.

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Comment by kevinsperrine** [comment]
@davidicus Can you re-trigger netlify on this one? The push failed when I updated stories to add the `hasFastSearch` knob.

**Review by davidicus — APPROVED** [review]
Great job!

File: packages/react/src/components/Table/TableToolbar/TableToolbar.jsx [hunk 1]
```diff
@@ -286,9 +291,18 @@ const TableToolbar = ({
               translateWithId={(...args) => tableTranslateWithId(i18n, ...args)}
               id={`${tableId}-toolbar-search`}
               onChange={(event, defaultValue) => {
-                // https://github.com/carbon-design-system/carbon/issues/6157
-                onApplySearch(event?.target?.value || defaultValue);
+                const value = event?.target?.value || defaultValue;
```
Comment (bjornalm) [inline packages/react/src/components/Table/TableToolbar/TableToolbar.jsx:294]: I think that the "Clear search input" in TableToolbarSearch's updates the value to be an empty string and expects this onChange method to perform the actual searchApply to clear the filter. So for the new hasFastSearch:false the search will not be cleared when the "Clear search input" is clicked.

**Review by bjornalm — CHANGES_REQUESTED** [review]

File: packages/react/cypress/plugins/index.js [hunk 2]
```diff
@@ -80,6 +80,10 @@ const webpackConfig = {
           },
         ],
       },
+      {
+        test: /\.mdx?$/,
```
[the remark this answers is not in the document]
Comment (kevinsperrine) [reply packages/react/cypress/plugins/index.js:84]: b/c of the mdx files imported in the table stories, when pulling the component from it, webpack needs to know how to process the mdx documentation files.

**Comment by kevinsperrine** [comment]
closing. as this was originally branched from master, and not next. reopening from next.

**Pull request closed without merging**